### ISSUE TITLE
NodeJS 8.0 requirement

### ISSUE THREAD

**Issue by simondel** [Microsoft/tfs-cli#246]
Hi! I noticed that this package has been updated to require NodeJS 8.0 or higher. 

Why is this the case? I see that the sourcecode is TypeScript with ES5 as a target. Is there perhaps a dependency that forces 8.0 or higher?

**Comment by trevorsg** [comment]
I removed the dependency on Q since ES6 promises are built into Node now. However, only in version 8 have they added the [Util.promisify](https://nodejs.org/api/util.html#util_util_promisify_original) method, which replaces Q.nfcall and Q.nfapply, which are used extensively throughout. I bumped the version of TFX to 0.5.x so that anyone with a ~ or ^ dependency won't pull in the new version.

Node 8.9.3 is the current LTS release, so I would think consumers are able to upgrade. But it seems there has been some pushback, so I will look into writing my own promisify to not depend on NodeJS 8.
